Marlin/src/HAL/shared/Delay.h

```diff
@@ -58,49 +58,81 @@
 
   #else
 
-    // https://blueprints.launchpad.net/gcc-arm-embedded/+spec/delay-cycles
-
-    #define nop() __asm__ __volatile__("nop;\n\t":::)
+    #if defined(HAL_SYSTICK_VALUE) && defined(HAL_SYSTICK_RELOAD_VALUE)
+      FORCE_INLINE static void DELAY_CYCLES(const uint32_t nbTicks) {
```

Comment:
Not sure to understand this part of code, but usually to realize delays I write something like (I don't use arm usually so I'm not sure it can works):
```c++
starttick = ticks();
while (ticks()-starttick < requiredTicks)
  ;
```
and I use the 32bits overflow logic to be sure I always get a positive value to check

Reply:
systick is initialised with HAL_SYSTICK_RELOAD_VALUE and decrements each cycle until it get 0, and then restart to HAL_SYSTICK_RELOAD_VALUE.
So I need to keep track of underflows that occur in the loop. The code is almost a copy of the smt32duino busy loop.